int8 data1
int16 data2
int32 data3